frontend/src/common/components/mentoringStepper/MentoringStepper/MentoringStepper.tsx

```diff
@@ -0,0 +1,183 @@
+import styled from '@emotion/styled';
+
+import tooltipIcon from '../../../assets/images/tooltip.svg';
+import {
+  MentoringReservationStatusTypeEnum,
+  type MentoringReservationStatusType,
+} from '../../../types/mentoringReservationStatus';
+import Step from '../Step/Step';
+
+interface MentoringStepperProps {
+  status: MentoringReservationStatusType;
+}
+
+const statusTextMap = Object.values(MentoringReservationStatusTypeEnum).reduce(
+  (acc, key) => {
+    switch (key) {
+      case 'PENDING':
+        acc[key] = '예약신청';
+        break;
+      case 'APPROVED':
+        acc[key] = '신청확정';
+        break;
+      case 'COMPLETE':
+        acc[key] = '멘토링완료';
+        break;
+    }
+    return acc;
+  },
+  {} as Record<string, string>,
+);
+
+const statusInfoTextMap: Record<MentoringReservationStatusTypeEnum, string> = {
+  PENDING: '예약 확정 완료 시 문자로 연락용 오픈카톡방 링크가 발송됩니다.',
+  APPROVED:
+    '신청이 확정되었습니다. 문자로 발송된 오픈카톡 링크를 통해 멘토와 대화를 시작할 수 있습니다.',
+  COMPLETE: '멘토링이 완료되었습니다.',
+};
+
+function MentoringStepper({ status }: MentoringStepperProps) {
+  const stepValues = Object.values(MentoringReservationStatusTypeEnum);
+
+  const stepOrder = stepValues.map((value, index) => ({
+    step: index,
+    status: value,
+  }));
+
+  const matchedStatus = stepOrder.find((item) => item.status === status);
+  const currentStep = matchedStatus ? matchedStatus.step : 0;
```

Comment:
```suggestion
    const currentStep = stepValues.indexOf(status);
```

p3: 이렇게 간략하게 바꿀수 있을 것 같은데 어때?

Reply:
index가 필요한 상황이 아니라, matchedStatus에서 status가 일치하는 요소의 step 값을 가져와야 해서 지금 방식대로 했어!
한줄로 간결하게 표현하는거면 이렇게 바꿔도 될 거 같아🙂

```
const currentStep = stepOrder.find((item) => item.status === status)?.step ?? 0;
```

Reply:
수정커밋: [refactor: 불필요한 객체 대신 배열형으로 변경하여 간결화](https://github.com/woowacourse-teams/2025-Fit-toring/pull/611/commits/e51ae78d133d0dbaf779258f0280a59c6bd2df21)

이렇게 하면 불필요하게 객체를 만들지 않아도 돼서 더 간결하게 돼! 이렇게 수정했어~~